src/main/java/step1/domain/DelimiterAndNumbers.java

```diff
@@ -0,0 +1,105 @@
+package step1.domain;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DelimiterAndNumbers {
+
+	private static final int ZERO = 0;
+
+	private static final Pattern SPACE = Pattern.compile("\\s");
+
+	private static final String EMPTY_STRING = "";
+
+	private static final Pattern DEFAULT_DELIMITER = Pattern.compile("[,:]");
+
+	private static final Pattern NON_CUSTOM_DELIMITER_PATTERN = Pattern.compile("(\\d"+DEFAULT_DELIMITER+")*\\d");
+
+	private static final Pattern CUSTOM_DELIMITER_PATTERN = Pattern.compile("//(.)\\\\n((\\d\\1)*\\d)");
+
+	private static final Pattern NON_NUMBER_PATTERN = Pattern.compile("\\D");
+
+	private final Pattern delimiter;
+
+	private final int[] numbers;
+
+	public DelimiterAndNumbers(String input) {
+		String removedSpace = removeSpace(input);
+		this.delimiter = initDelimiter(removedSpace);
+		this.numbers = initNumbers(removedSpace);
+	}
+	//For Test
+	public DelimiterAndNumbers(Pattern delimiter, int[] numbers) {
+		this.delimiter = delimiter;
+		this.numbers = numbers;
+	}
+
+	public int sum() {
+		return Arrays.stream(numbers)
+				.reduce(ZERO, Integer::sum);
+	}
+
+	private static String removeSpace(String string) {
+		return SPACE.matcher(string)
+				.replaceAll(EMPTY_STRING);
+	}
```

Comment:
공백 패턴은 필요 없을 것 같아요 :)
```suggestion
	private static String removeSpace(String string) {
		return string.replaceAll("\\s", EMPTY_STRING);
	}
```
```
@DisplayName("공백 제거 테스트")
@ParameterizedTest
@ValueSource(strings = {" ", "  ", "    ", "a ", "a  ", " a "})
void removeSpaceTest(String space) {
    String result = space.replaceAll("\\s", "");
    assertThat(result).doesNotContainAnyWhitespaces();
}
```

Reply:
패턴을 미리 캐시해놓으려고 만들어 뒀던 것인데, 캐시를 없애는걸 제안해 주신 이유가 있을까요??

Reply:
패턴 캐싱에 대한 접근은 좋다고 생각합니다 👍 
다만 현재 많은 패턴 캐싱하여 관리하고 있어 줄일 필요가 있다고 생각했습니다.
말씀하신대로 현재 상태를 유지하셔도 됩니다 :)